Dart/src/com/jetbrains/lang/dart/ide/template/postfix/DartRemotePostfixTemplate.java

```diff
@@ -0,0 +1,195 @@
+package com.jetbrains.lang.dart.ide.template.postfix;
+
+import com.intellij.CommonBundle;
+import com.intellij.codeInsight.template.postfix.templates.PostfixTemplate;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.refactoring.util.CommonRefactoringUtil;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService;
+import com.jetbrains.lang.dart.assists.AssistUtils;
+import com.jetbrains.lang.dart.assists.DartSourceEditException;
+import org.dartlang.analysis.server.protocol.Position;
+import org.dartlang.analysis.server.protocol.PostfixCompletionTemplate;
+import org.dartlang.analysis.server.protocol.SourceChange;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Set;
+
+public class DartRemotePostfixTemplate extends PostfixTemplate {
+
+  protected DartRemotePostfixTemplate(PostfixCompletionTemplate template) {
+    this(template.getName(), template.getKey(), template.getExample());
+  }
+
+  protected DartRemotePostfixTemplate(@NotNull String name, @NotNull String key, @NotNull String example) {
+    super(name, key, example);
+  }
+
+  public static DartRemotePostfixTemplate createTemplate(PostfixCompletionTemplate template) {
+    // TODO Automatically generate documentation subclasses from server data during build.
+    try {
+      String prefix = StringUtil.capitalize(template.getName());
+      if (prefix.equals("!")) {
+        prefix = "Bang";
+      }
+      Class <?> templateClass = Class.forName("com.jetbrains.lang.dart.ide.template.postfix." + prefix + "PostfixTemplate");
+      Constructor<?> constructor = ((Class<?>)templateClass).getDeclaredConstructor(PostfixCompletionTemplate.class);
+      return (DartRemotePostfixTemplate)constructor.newInstance(template);
+    }
+    catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InvocationTargetException | InstantiationException ex) {
+      return new DartRemotePostfixTemplate(template);
+    }
+  }
+
+  @Override
+  public boolean isApplicable(@NotNull PsiElement context, @NotNull Document copyDocument, int newOffset) {
+    final Project project = context.getProject();
+    final PsiFile psiFile = context.getContainingFile();
+    final DartAnalysisServerService service = DartAnalysisServerService.getInstance(project);
+    String version = service.getSdkVersion();
+    Set<PostfixTemplate> templates = DartPostfixTemplateProvider.getTemplates(version);
+    boolean found = false;
+    for (PostfixTemplate temp : templates) {
```

Comment:
I've got an NPE here when switched to a different project with older Dart SDK. Marked `DartPostfixTemplateProvider.getTemplates(version)` as `@Nullable` and added null check here.